nylas/handler/http_client.py

```diff
@@ -126,9 +130,9 @@ def _execute_download_request(
 
             # If we stream an iterator for streaming the content, otherwise return the entire byte array
             if stream:
-                return response
+                return _validate_download_response(response,stream)
```

Comment:
Maybe we can simplify this?

```suggestion
                return response if response.ok else response.content
```

Reply:
But then, will `_execute_download_request` not error out where ever it is used 🤔 So I guess the question is, is this the correct place to do this validate this?